sqlglot/parser.py

```diff
@@ -668,18 +655,14 @@ def _parse_order(self):
 
     def _parse_ordered(self):
         this = self._parse_bitwise()
-        desc = self._match(TokenType.ASC) or self._match(TokenType.DESC)
-        return self.expression(
-            exp.Ordered,
-            this=this,
-            desc=desc.token_type is TokenType.DESC if desc else False,
-        )
+        self._match(TokenType.ASC)
+        return self.expression(exp.Ordered, this=this, desc=self._match(TokenType.DESC))
 
     def _parse_limit(self):
         if not self._match(TokenType.LIMIT):
             return None
 
-        limit_number = self._match(TokenType.NUMBER)
+        limit_number = self._match(TokenType.NUMBER) and self._prev
```

Comment:
Given that we previously matched the `LIMIT` token how can `self._prev` be None in this case?

Reply:
Never mind. I confused it with `_curr`.